libmambapy/CMakeLists.txt

```diff
@@ -32,6 +32,14 @@ pybind11_add_module(
     src/libmambapy/bindings/solver.cpp
     src/libmambapy/bindings/solver_libsolv.cpp
 )
+# TODO: remove when `SubdirData::cache_path()` is removed
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+    # This file uses capturing structured bindings, which was fixed in C++20
+    set_source_files_properties(
+        src/libmambapy/bindings/legacy.cpp
+        PROPERTIES COMPILE_FLAGS -Wno-error=deprecated-declarations
```

Comment:
I turned this back to warning. Sometimes it's useful to see when we're using deprecated functions, so let's have it as a warning.

Reply:
It makes sense for that code or even the whole library. 👍🏽

Reply:
For now, let's keep it for this file. If needed later, we can add this flag to the whole repo/lib I suppose.
It's better not to use deprecated functions, when possible.